fs/operations.go

```diff
@@ -1588,6 +1588,24 @@ func Rcat(fdst Fs, dstFileName string, in0 io.ReadCloser, modTime time.Time) (er
 	defer func() {
 		Stats.DoneTransferring(dstFileName, err == nil)
 	}()
+	defer in0.Close()
+
+	hashOption := &HashesOption{Hashes: NewHashSet()}
+
+	in := in0
+	buf := make([]byte, 100*1024)
+	if n, err := io.ReadFull(in0, buf); err != nil {
```

Comment:
io.ReadFull is tricky to use correctly!

Here you are ignoring if it returned a real error.  so you probably need a

```go
if err != io.EOF && err != io.ErrUnexpectedEOF {
    return err
}
```